### DISCUSSION TITLE
help installing on macOS

### DISCUSSION THREAD

**Discussion by dnews13** [VioletteChiara/AnimalTA#11] [Q&A] [answered]
I've been using the program on Windows with exceptional ease. However, I'm running into issues getting the program installed on macOS. I've referred to the Guidelines .pdf but I don't see any install help in there.

So far, I've attempted to clone the repository and use pip3 to install the program. However, I am running into some errors:
`ERROR: git+https://github.com/VioletteChiara/AnimalTA.git does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.`

After that, I changed `cli.py` to `setup.py` to see if that would help. Then, I went into the cloned folder itself and tried installing from there and this error was produced: `ModuleNotFoundError: No module named 'pyautogui'`. I suspect the cascade of changing the name may have had something to do with that, but I am unsure of where to go from here.

Thanks in advance, and thank you for the work on such a useful analysis program!

**Comment by VioletteChiara** [comment, marked as answer]
Hello, 

Sorry for the delay in my answer. Unfortunately as there is no installer for Mac yet, you must install each separate packages used by the program.

To do so, you can use the command: "pip3 install name_of_the_package"
For instance, in that case, you will have to write: pip3 install PyAutoGUI.

Here is a list of the required packages:
decord
matplotlib
MouseInfo
numpy
opencv-python
packaging
PIL-Tools
Pillow
playsound
psutil
PyAutoGUI
sklearn
beepy
pandas
scikit-learn


Kind regards!

**Reply by al356lock** [reply]
hi @dnews13 did you get this working on your mac in the end? I am new to pyhton as I use R but it seems as though some of the packages are not installing e.g. decord and it looks like other mac users have had similar problems? Thanks

**Reply by VioletteChiara** [reply]
Hi, you can try to use eva-decord instaed of decord if decord is not available. Howver, a simpler solution would be to use virtualization software. We have good feedback from users using UTM (https://mac.getutm.app/) or Parallel Desktop (https://www.parallels.com/products).

**Reply by al356lock** [reply]
Hi Violette, 

I have managed to download the software now and have followed the tutorials on YouTube. I have prepared the video for tracking and pressed begin tracking. However, it has been running for 3 hours and is still not half way.  On your tutorial it seems to load very quickly, but I assume the video you used to track might be very short. Mine are 30 minutes long. I was wondeirng if this processing time is unusual or is this normal for the length of the video? Were you using a computer for yours, I was thinking it could perhaps be because I am using my laptop?

Thank you

**Reply by VioletteChiara** [reply]
Hi!

Six hours for a 30min video is indeed very slow. In my case, I also use a laptop and it is indeed much faster in the very large majority of cases.

However, the tracking duration is very variable between videos.

Here are a few characteristics that might slow down your tracking, you may try to change some of them:
- Frame rate: the higher the frame rate, the slower the tracking will be
- Video resolution: Videos recorded in high resolution are slower to process, 4K videos are for instance very slow.
- Number of individuals to track: the more numerous are your targets the slower will be the tracking, it gets even slower if you have several targets in the same arena.
- The parameters:
     --> Stabilisation, light correction, flicker correction, color tracking, and relative background are time-consuming options
     --> **Size filtering is very expensive in time**, my advice to get the tracking as fast as possible is to avoid having too much noise before that step, you may play with the previous parameters (threshold, erosion, dilation) to minimize the number of non-targets white blobs.

**Reply by al356lock** [reply]
Hi Violette, 

I lowered the resolution and it is down to 2 minutes per video, so pleased. 

Thank you for creating this programme, it has been so helpful with my masters project! 

Alice

**Reply by VioletteChiara** [reply]
Hi Alice,

Thank you for trying AnimalTA! I am glad it is now working properly and I wish you all the best for your master project!

Violette
